### DISCUSSION TITLE
`EntityRepository.create` method must be async

### DISCUSSION THREAD

**Discussion by ruscon** [mikro-orm/mikro-orm#4716] [General]
**Describe the bug**
if the business logic is executed in a cli script, then the last record does not have time to be recorded.

Due to the fact that the `persistOnCreate` option is enabled by default, it was not immediately clear why this was happening.

**Expected behavior**
1. `EntityRepository.create` must be async
2. Probably `persistOnCreate` shouldn't be enabled by default

**Versions**
5.8.1

**Comment by B4nan** [comment]
Not sure what your problem exactly is, but I don't see how `em.create` being sync or the `persistOnCreate` could have any effect on that.

If you provide a complete repro, I will be happy to have a closer look.

**Reply by ruscon** [reply]
I'm sorry I was wrong.
I had a problem because of `FlushMode.AUTO`
